src/components/Volunteer/CoreVolunteer.jsx

```diff
@@ -0,0 +1,241 @@
+import React from "react";
+import Select from "react-select";
+
+import { addCoreVolunteer, getCoreVolunteers } from "../../services/volunteerService";
+import { Input, Button } from "../../commons/Form";
+import { getColleges } from "../../services/collegeServices";
+import { toast } from "../../actions/toastActions";
+
+const sizes = [
+
+    { value: 's', label: 'Small' },
+    { value: 'm', label: 'Medium' },
+    { value: 'l', label: 'Large' },
+    { value: 'xl', label: 'X Large' },
+];
+
+class CoreVolunteer extends React.Component {
+    ADD_VOLUNTEER = "Add";
+    state = {
+        buttonText: this.ADD_VOLUNTEER,
+        name: "",
+        registerNumber: "",
+        shirtSize: null,
+        college: null,
+        volunteers: []
+    };
+
+    componentWillMount() {
+        this.getColleges();
+        this.getVolunteers();
+    }
+
+    getColleges = async () => {
+        let response = await getColleges();
+        if (response.status === 200) {
+            const colleges = response.data.map(college => ({
+                value: college.id,
+                label: college.name + ", " + college.location
+            }))
+            this.setState({ colleges })
+        }
+        else
+            toast("An error occured while fetching colleges");
+    }
+
+    handleChange = (e) => {
+        this.setState({
+            [e.name]: e.value,
+        });
+    };
+
+    addVolunteer = async () => {
+        try {
+            await this.setState({ buttonText: this.ADDING_VOLUNTEER });
+            const { name, registerNumber, shirtSize, college } = this.state;
+            if (!name || name.length === 0)
+                throw Error("Please enter name.");
+            if (!registerNumber || registerNumber.length === 0)
+                throw Error("Please enter register number.");
+            if (!shirtSize || shirtSize.length === 0)
+                throw Error("Please select shirt size.");
+            if (!college || college.length === 0)
+                throw Error("Please select the college.");
+            let response = await addCoreVolunteer({
+                name,
+                registerNumber,
+                shirtSize,
+                college
+            });
+            this.setState({
+                buttonText: this.ADD_VOLUNTEER,
+            })
+            toast(response.message);
+            this.getVolunteers();
+            this.setState({ name: "", registerNumber: "" })
```

Comment:
Do you think you can handle this with a single setState call
FYI another setState being called from getVolunteers

Reply:
this.getVolunteers is a fetch request and is called multiple times

https://github.com/manipalutsav/frontend/blob/038771a0ae7f33a7b2611dbe60b6ef87777782f9/src/components/Volunteer/CoreVolunteer.jsx#L28-L31

Besides, moving this setState inside might make the name & registerNumber values set to empty when the user is typing the value or submitting if this.getVolunteers is fetching meanwhile.

Reply:
What's the point of this `setState`? It doesn't change the default state, so what is it doing?

Reply:
setState is resetting the name & registernumber to blank so that the user can enter another volunteer. If you check the UI you'll get to know the purpose of it.